processor/deltatorateprocessor/processor.go

```diff
@@ -41,10 +49,64 @@ func (dtrp *deltaToRateProcessor) Start(context.Context, component.Host) error {
 
 // processMetrics implements the ProcessMetricsFunc type.
 func (dtrp *deltaToRateProcessor) processMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+	resourceMetricsSlice := md.ResourceMetrics()
+
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		rm := resourceMetricsSlice.At(i)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for i := 0; i < ilms.Len(); i++ {
+			ilm := ilms.At(i)
+			metricSlice := ilm.Metrics()
+			for j := 0; j < metricSlice.Len(); j++ {
+				metric := metricSlice.At(j)
+				_, ok := dtrp.metrics[metric.Name()]
```

Comment:
Does this apply to all metrics? Do we need a way to specify which metrics to apply to?

Reply:
We are getting the metric names from the config file and mathing them before applying the conversions. I am not sure if I answered your question. Can you explain a bit if I missed your point.

Reply:
This is because the code is written opposite to the rest of the code. It should be:

```
if !ok {
  continue
}
...
```

This also removes one indentation.

Reply:
Oh I see. You are smart. Will update.

Reply:
Also `metrics` member name is bad, does not suggest anything about "filtering" :)

Reply:
Pushed an update.